libcontainer/nsenter/nsexec.c

```diff
@@ -741,6 +754,22 @@ void nsexec(void)
 					bail("failed to dup stderr");
 			}
 
+			/* ... wait until our topmost parent has finished cgroup setup in p.manager.Apply() ... */
+			if (config.cloneflags & CLONE_NEWCGROUP) {
+				len = read(pipenum, buf, SYNC_BUF_LEN);
+				if (len < 0)
+					bail("read synchronisation value failed");
+				value = strtol(buf, &endptr, 10);
+				if (*endptr != '\0')
+					bail("unable to parse synchronisation value");
+				if (value == SYNC_PRIVATE_CGNS) {
+					if (unshare(CLONE_NEWCGROUP) < 0)
```

Comment:
I think that RHEL might have issues with this (WDYT @mrunalp). Also this is a _bit_ ugly but I'm guessing you can't combine this with the rest of the `unshare` code because we're currently sending the PID of `JUMP_INIT` not `JUMP_CHILD` to the `.Apply` code